rest_framework/serializers.py

```diff
@@ -268,7 +269,7 @@ def _get_declared_fields(cls, bases, attrs):
             if hasattr(base, '_declared_fields'):
                 fields = list(base._declared_fields.items()) + fields
 
-        return OrderedDict(fields)
+        return attrs.get('_dict_class', OrderedDict)(fields)
```

Comment:
My feeling is that if we did use an overridable dict, then we should probably just use it from the return data / error data, rather than for the fields themselves.